contrib/utilities/indent_common.sh

```diff
@@ -0,0 +1,167 @@
+# temporary file and diff as a workaround.
+#
+
+format_file()
+{
+  clang-format "$1" > "$1.tmp${NONCE}"
+  diff -q "$1" "$1.tmp${NONCE}" >/dev/null || mv "$1.tmp${NONCE}" "$1"
+  rm -f "$1.tmp${NONCE}"
+}
+export -f format_file
+
+#
+# In order to format .inst.in files, we need to replace \{ and \} by a
+# sentinel because clang-format happily strips away the backslash. Further,
+# there is a minor caveat: clang-format automatically renames a comment
+# after a closing bracket for a namespace to "} // namespace FooBar". Thus
+# use "namespace" as keyword:
+#
+
+format_inst()
+{
+    cp "$1" "$1.tmp${NONCE}"
+    sed -i 's#\\{#{ // namespace#g' "$1.tmp${NONCE}"
+    sed -i 's#\\}#} // namespace#g' "$1.tmp${NONCE}"
+    clang-format -i "$1.tmp${NONCE}"
+    sed -i 's#{ // namespace#\\{#g' "$1.tmp${NONCE}"
+    sed -i 's#}[ ]*// namespace.*#\\}#g' "$1.tmp${NONCE}"
+
+    diff -q "$1" "$1.tmp${NONCE}" >/dev/null || mv "$1.tmp${NONCE}" "$1"
+    rm -f "$1.tmp${NONCE}"
+}
+export -f format_inst
+
+#
+# Convert DOS formatted files to unix file format by stripping out
+# carriage returns (15=0x0D):
+#
+
+dos_to_unix()
+{
+  tr -d '\015' <"$1" >"$1.tmp${NONCE}"
+  diff -q "$1" "$1.tmp${NONCE}" >/dev/null || mv "$1.tmp${NONCE}" "$1"
+  rm -f "$1.tmp${NONCE}"
+}
+export -f dos_to_unix
+
+#
+# Fix permissions
+#
+
+fix_permissions()
+{
+  chmod 644 "$1"
+}
+export -f fix_permissions
+
+#
+# Collect all files found in a list of directories "${1}$" matching a
+# regular expression "${2}$", and process them with a command "${3}" on 10
+# threads in parallel.
+#
+# The command line is a bit complicated, so let's discuss the more
+# complicated arguments:
+# - For 'find', -print0 makes sure that file names are separated by \0
+#   characters, as opposed to the default \n. That's because, surprisingly,
+#   \n is a valid character in a file name, whereas \0 is not -- so it
+#   serves as a good candidate to separate individual file names.
+# - For 'xargs', -0 does the opposite: it separates filenames that are
+#   delimited by \0
+# - the options "-n 1 -P 10" make sure that the following script with be
+#   called exactly with one file name as argument at a time, but we allow
+#   execution for up to 10 times in parallel
+#
+
+process()
+{
+  find ${1} -regextype egrep -regex "${2}" -print0 |
+    xargs -0 -n 1 -P 10 -I {} bash -c "${3} {}"
+}
+
+#
+# Variant of above function that only processes files that have changed
+# since the last merge commit to master. For this, we collect all files
+# that
+#  - are new
+#  - have changed since the last merge commit to master
+#
+
+process_changed()
+{
+  LAST_MERGE_COMMIT="$(git log --format="%H" --merges --max-count=1 master)"
+
+  ( git ls-files -z --others --exclude-standard -- ${1};
+    git diff -z --name-only $LAST_MERGE_COMMIT -- ${1} )|
+      sort -zu |
+      grep -zE "${2}" |
+      xargs --no-run-if-empty -0 -n 1 -P 10 -I {} bash -c "${3} {}"
+}
```

Comment:
@drwells 
In light of #6762 , I took the liberty to reorganize your script as well.
The reorganization also made the whole `base64` magic superfluous, because we can now simply pipe the output directly into xargs :-)